napari/_vispy/layers/image.py

```diff
@@ -98,10 +98,7 @@ def _on_display_change(self, data=None):
         if data is None:
             data = np.zeros((1,) * ndisplay, dtype=np.float32)
 
-        if self.layer._empty:
```

Comment:
As a first pass I removed these because they are strongly connected to the way the old async works.

But there are still cases when we can have an empty slicing (i.e. no actual slicing has happened yet), but the vispy layer will still ask for the slice (e.g. when `_set_view_slice` returns early due to OOD indices).

So I should probably add this back. I also wonder if we can merge the concepts of `loaded` and `empty`.